src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx

```diff
@@ -215,17 +227,31 @@ const getDuplicatedScoresAndReports = (
   scoresAndReports?: ScoresAndReports,
 ) => {
   const reports = scoresAndReports?.reports?.map((report) => {
-    const conditionalLogic =
-      report.type === ScoreReportType.Score
-        ? report.conditionalLogic
-        : report.conditionalLogic && {
-            ...report.conditionalLogic,
-            conditions: getDuplicatedConditions(
-              oldItems,
-              newItems,
-              report.conditionalLogic?.conditions,
-            ),
-          };
+    let conditionalLogic;
+    if (report.type === ScoreReportType.Section) {
+      const scoreReports = scoresAndReports?.reports.filter(
```

Comment:
could you pls make filter and reduce in one mapper (one reduce for example) to increase productivity?